lib/tpm2_alg_util.c

```diff
@@ -300,3 +300,45 @@ bool pcr_parse_digest_list(char **argv, int len,
 
     return true;
 }
+
+bool tpm2_extract_plain_signature(UINT8 **buffer, UINT16 *size,
+    TPMT_SIGNATURE *signature) {
+
+    *buffer = NULL;
+    *size = 0;
+
+    switch (signature->sigAlg) {
+    case TPM_ALG_RSASSA:
+        *size = sizeof((*signature).signature.rsassa.sig.t.buffer);
+        *buffer = malloc(*size);
```

Comment:
validate malloc return here and throughout

Reply:
Just return buffer instead of the bool.

Reply:
I've adjusted this accordingly.